lib/auth/linkedin_oidc_authenticator.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class Auth::LinkedInOidcAuthenticator < Auth::ManagedAuthenticator
+  class LinkedInOidc < OmniAuth::Strategies::OAuth2
+    option :name, "linkedin_oidc"
+
+    option :client_options,
+           {
+             site: "https://api.linkedin.com",
+             authorize_url: "https://www.linkedin.com/oauth/v2/authorization?response_type=code",
+             token_url: "https://www.linkedin.com/oauth/v2/accessToken",
+           }
+
+    option :scope, "openid profile email"
+
+    uid { raw_info["sub"] }
+
+    info do
+      {
+        email: raw_info["email"],
+        first_name: raw_info["given_name"],
+        last_name: raw_info["family_name"],
+        image: raw_info["picture"],
+      }
+    end
+
+    extra { { "raw_info" => raw_info } }
+
+    def callback_url
+      full_host + script_name + callback_path
+    end
+
+    alias oauth2_access_token access_token
+
+    def access_token
+      ::OAuth2::AccessToken.new(
+        client,
+        oauth2_access_token.token,
+        {
+          expires_in: oauth2_access_token.expires_in,
+          expires_at: oauth2_access_token.expires_at,
+          refresh_token: oauth2_access_token.refresh_token,
+        },
+      )
+    end
```

Comment:
What's the reason for doing this instead of using the `access_token` set up by OmniAuth::OAuth2? 🤔